File: kotlin/resistor-color/src/main/kotlin/ResistorColor.kt
object ResistorColor {

    private val Colors = listOf("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white")

    // TODO: error handling for invalid color
    fun colorCode(input: String): Int = Colors.indexOf(input)

    fun colors(): List<String> = Colors
}

File: typescript/resistor-color/resistor-color.ts
export const COLORS: string[] = [
  'black', 'brown', 'red', 'orange', 'yellow', 
  'green', 'blue', 'violet', 'grey', 'white'
];

export const colorCode = (color: string): number => {
  const code = COLORS.indexOf(color);
  if (code == -1)
    throw new Error("Invalid color: " + color);
  return code;
}
